tests/PySys/environments/environment_c8y.py

```diff
@@ -0,0 +1,117 @@
+import pysys
+from pysys.basetest import BaseTest
+
+"""
+Environment to manage automated connect and disconnect to c8y
+
+Tests that derive from class EnvironmentC8y use automated connect and
+disconnect to Cumulocity. Additional checks are made for the status of
+service mosquitto and service tedge-mapper.
+"""
+
+
+class EnvironmentC8y(BaseTest):
+    def setup(self):
+
+        self.tedge = "/usr/bin/tedge"
+        self.sudo = "/usr/bin/sudo"
+        self.systemctl = "/usr/bin/systemctl"
+        self.log.info("EnvironmentC8y Setup")
+        self.addCleanupFunction(self.mycleanup)
+
+        # Check if tedge-mapper is in active
+        serv_mapper = self.startProcess(
+            command=self.systemctl,
+            arguments=["status", "tedge-mapper"],
+            stdouterr="serv_mapper1",
+            expectedExitStatus="==3", # 3: disabled
+        )
+
+        if serv_mapper.exitStatus != 3:
+            self.log.error("The tedge-mapper service is running")
+            self.abort(FAILED)
+
+        # Connect the bridge
+        connect = self.startProcess(
+            command=self.sudo,
+            arguments=[self.tedge, "connect", "c8y"],
+            stdouterr="tedge_connect",
+            expectedExitStatus="==0",
+        )
+
+        # Check if mosquitto is running well
+        serv_mosq = self.startProcess(
+            command=self.systemctl,
+            arguments=["status", "mosquitto"],
+            stdouterr="serv_mosq2",
+            expectedExitStatus="==0",
+        )
+
+        if serv_mosq.exitStatus != 0:
```

Comment:
Same comment as above. If the `exitStatus` is non-zero, the previous statement itself will fail.